lib/modules/manager/gomod/artifacts.ts

```diff
@@ -159,11 +159,23 @@ export async function updateArtifacts({
   const useVendor = (await readLocalFile(vendorModulesFileName)) !== null;
 
   let massagedGoMod = newGoModContent;
-
   if (!config.postUpdateOptions?.includes('gomodNoMassage')) {
     // Regex match inline replace directive, example:
     // replace golang.org/x/net v1.2.3 => example.com/fork/net v1.4.5
     // https://go.dev/ref/mod#go-mod-file-replace
+
+    // replace bracket after comments, so it doesn't break the regex, doing a complex regex causes problems
+    // when there's a comment and ")" after it, the regex will read replace block until comment.. and stop.
+    massagedGoMod = massagedGoMod
+      .split('\n')
+      .map((line) => {
+        if (line.trim().startsWith('//')) {
+          return line.replace(')', 'renovate-replace-bracket');
```

Comment:
Pretty hacky but I guess it works?

Reply:
Yes, that's how its done line by line :D you wanted a fast fix! there you go...
The other way is to do a complex regex but its problematic.